src/function/vector_cast_functions.cpp

```diff
@@ -69,12 +69,18 @@ static void resolveNestedVector(std::shared_ptr<ValueVector> inputVector, ValueV
     }
 
     // non-nested types
-    scalar_func_exec_t func = CastFunction::bindCastFunction<CastChildFunctionExecutor>(
-        "CAST", inputType->getLogicalTypeID(), resultType->getLogicalTypeID())
-                                  ->execFunc;
-    std::vector<std::shared_ptr<ValueVector>> childParams{inputVector};
-    dataPtr->numOfEntries = numOfEntries;
-    func(childParams, *resultVector, (void*)dataPtr);
+    if (inputType->getLogicalTypeID() != resultType->getLogicalTypeID()) {
```

Comment:
assert inputVector is `unFiltered` before line 72.

Reply:
When run locally, the assertion works just fine, but when run in the CI, the code segfaults on tests. It's possible that the assertion fails on different platforms. Inspecting the previous code, there was no unfiltered assertion previously. It seems like this code was written without the filter in mind, so I removed the assert.